### PR TITLE
GEN-1909 new SDK function - set_secret

### PR DESCRIPTION
**Pull request by hrgreene**
## Change description

> New SDK function, allows process running on wayscript to create or update a secret
```
from wayscript import secret_manager

secret_manager.set_secret("key", "value")
```

## Checklist

- [x] Application changes have been tested and automated tests have been added, if applicable
- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

## Related issues

> Fixes #1909

### CODE REVIEW

**Review by samkam — APPROVED** [review]
LGTM

File: src/wayscript/secret_manager.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    lair_id = process_details["lair_id"]
+    client = utils.WayScriptClient()
+    response = client.set_lair_secret(lair_id, secret_key, secret_val)
+    
+    # Handle unhelpful error cases
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        if e.code == 403:
+            raise errors.UnauthorizedUserError
```
Comment (jeffbuswell) [inline src/wayscript/secret_manager.py]: do you need the constructor parentheses here? `errors.UnauthorizedUserError()`

Comment (aroxby-wayscript) [reply src/wayscript/secret_manager.py]: Fun fact, no.
https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

File: src/wayscript/secret_manager.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from urllib.error import HTTPError
+
+from . import utils, context, errors
+
+def set_secret(secret_key: str, secret_val: str) -> bool:
```
Comment (aroxby-wayscript) [inline src/wayscript/secret_manager.py]: You might want to run flake8 on the files changed.
```suggestion


def set_secret(secret_key: str, secret_val: str) -> bool:
```

Comment (aroxby-wayscript) [inline src/wayscript/secret_manager.py]: Are you sure you want this to return a bool since it is always true?

File: tests/test_secret_manager.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+from wayscript import secret_manager as ws_secret_manager
+
+
+@responses.activate
+def test_set_secret(patch_client_get_url):
+    """Test set secret"""
+    responses.add(responses.POST, patch_client_get_url, json={}, status=200)
+
+    callable = getattr(ws_secret_manager, 'set_secret')
+    assert callable("test_key", "test_val") == True
```
Comment (aroxby-wayscript) [inline tests/test_secret_manager.py]: What's the difference between this and
```
result = ws_secret_manager.set_secret('test_key', 'test_val')
assert result == True
```

**Review by aroxby-wayscript — COMMENTED** [review]
Really nice!  A little cleanup suggested.

File: tests/test_secret_manager.py (hunk 3 above)
Comment (hrgreene) [reply tests/test_secret_manager.py]: Probably identical, I was following the pattern set by the rest of the SDK's tests.

File: src/wayscript/secret_manager.py (hunk 2 above)
Comment (hrgreene) [reply src/wayscript/secret_manager.py]: Ah, yeah, there's no lint settings file in the repo so it wasn't enabled. I'll take a linting pass

File: tests/test_secret_manager.py (hunk 3 above)
Comment (aroxby-wayscript) [reply tests/test_secret_manager.py]: Feel free to keep it but I don't think this a pattern we want continue using.  The best example I found of this was in [test_context.py](https://github.com/wayscript/wayscript-python/blob/55abcf4f22869fccf6dbc7853444b70339683f2a/tests/test_context.py) which appears do it both ways.

File: src/wayscript/secret_manager.py (hunk 2 above)
Comment (hrgreene) [reply src/wayscript/secret_manager.py]: haha, used to a language where exceptions Didn't Exist so returning True was for successes and False for failure, hence typing this sort of operation as boolean...

Changed to None

File: src/wayscript/secret_manager.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    lair_id = process_details["lair_id"]
+    client = utils.WayScriptClient()
+    response = client.set_lair_secret(lair_id, secret_key, secret_val)
+
+    # Handle unhelpful error cases
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        if response.code == 403:
+            raise errors.UnauthorizedUserError from None
```
Comment (aroxby-wayscript) [inline src/wayscript/secret_manager.py]: Cool! Today I learned `raise ... from ...`

File: tests/test_secret_manager.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+@responses.activate
+def test_set_secret_unauthorized_user(patch_client_get_url):
+    """Test that set secret raises the appropriate error for unauthorized users"""
+    responses.add(responses.POST, patch_client_get_url, json={}, status=403)
+
+    try:
+        secret_manager.set_secret("test_key", "test_val")
+    except UnauthorizedUserError:
+        assert True
```
Comment (aroxby-wayscript) [inline tests/test_secret_manager.py]: If `set_secret` doesn't raise an exception this test will still pass.  Try [`pytest.raises`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises).

**Review by aroxby-wayscript — CHANGES_REQUESTED** [review]

**Review by aroxby-wayscript — APPROVED** [review]
LGTM!

**Pull request merged**